src/nitpackage.nit

```diff
@@ -83,6 +92,77 @@ private class ReadmePhase
 		toolcontext.info("{mpackage} moved to {path}", 0)
 	end
 
+	var allowed_ini_keys: Array[String] = [
+		"package.name", "package.desc", "package.tags", "package.license",
+		"package.maintainer", "package.more_contributors",
+		"upstream.browse", "upstream.git", "upstream.git.directory",
+		"upstream.homepage", "upstream.issues"
+		]
+
+	fun check_ini(mpackage: MPackage) do
+		if not mpackage.has_ini then
+			toolcontext.error(mpackage.location, "No `package.ini` file for `{mpackage}`")
+			return
+		end
+
+		var pkg_path = mpackage.package_path
+		if pkg_path == null then return
+
+		var ini_path = mpackage.ini_path
+		if ini_path == null then return
+
+		var ini = new ConfigTree(ini_path)
+
+		check_ini_key(mpackage, ini, "package.name", mpackage.name)
+		# check_ini_key(mpackage, ini, "package.desc")
+		check_ini_key(mpackage, ini, "package.tags")
+
+		# FIXME since `git reflog --follow` seems bugged
+		check_ini_key(mpackage, ini, "package.maintainer")
+		# var maint = mpackage.maintainer
+		# if maint != null then
+			# check_ini_key(mpackage, ini, "package.maintainer", maint)
+		# end
+
+		# FIXME since `git reflog --follow` seems bugged
+		# var contribs = mpackage.contributors
+		# if contribs.not_empty then
+			# check_ini_key(mpackage, ini, "package.more_contributors", contribs.join(", "))
+		# end
+
+		check_ini_key(mpackage, ini, "package.license", mpackage.license)
+		check_ini_key(mpackage, ini, "upstream.browse", mpackage.browse_url)
+		check_ini_key(mpackage, ini, "upstream.git", mpackage.git_url)
+		check_ini_key(mpackage, ini, "upstream.git.directory", mpackage.git_dir)
+		check_ini_key(mpackage, ini, "upstream.homepage", mpackage.homepage_url)
```

Comment:
You should assume the homepage is correct since if it is modified by the user it is the one you want to keep.
Maybe just check that the page does not 404 ?

Reply:
Since the `maintainer` and `more_contributors` checks will not work, you still have to manually commit the accepted changes.

So, I don't think we need to bother about really checking what is going on in the ini.